src/peft/tuners/lora/config.py

```diff
@@ -117,6 +118,25 @@ class LoraConfig(PeftConfig):
             )
         },
     )
+    bits_pattern: Optional[dict] = field(
+        default_factory=dict,
+        metadata={
+            "help": (
+                "The mapping from layer names or regexp expression to ranks which are different from the default bits specified by `bits`. "
+                "For example, `{model.decoder.layers.0.encoder_attn.k_proj: 2`}"
+            )
+        },
+    )
+    loftq_bits: Optional[int] = field(default=4, metadata={"help": "Quantization bits for LoftQ"})
+    loftq_iter: Optional[int] = field(default=1, metadata={"help": "Alternating iterations for LoftQ"})
+    loftq_fake: Optional[bool] = field(
+        default=True,
+        metadata={
+            "help": "True: use fp16/fp32; used for first time to save weights."
+            "False: use bitsandbytes 4bit linear models. weights can't be saved."
+            "Recommend to set to True, save the weights and load the saved weights in 4 bits."
```

Comment:
Could you please explain why this is needed? What weights are we talking about? Usually, we only save the adapter weights, not the weights of the original model.